TESTING.md

```diff
@@ -54,6 +54,40 @@ We recommend that you start the emulator on the remote machine using the [Google
 gcloud beta emulators datastore start --host-port <hostname of machine>:<port>
 ```
 
+### Testing code that uses DNS
+
+#### On your machine
+
+You can test against a in-memory local DNS by following these steps:
+
+1. Before running your testing code, start the DNS emulator `LocalDnsHelper`. This can be done as follows:
+
+  ```java
+  long delay = 0;
+  LocalDnsHelper helper = LocalDnsHelper.create(delay);
+  helper.start();
+  ```
+
+  This will spawn a server thread that listens to `localhost` at an ephemeral port for DNS requests.
+  The `delay` parameter determines if change requests should be processed synchronously
+  (value `0`) or in a separate thread with a delay of `delay` milliseconds.
+
+2. In your program, create and use a DNS service object by using the `options()` of `LocalDnsHelper`.  For example:
```

Comment:
Alternative, if you want:
In your program, create the DNS service by using the helper's `options()` method.